### DISCUSSION TITLE
Need explicit explanation of the environment dependencies

### DISCUSSION THREAD

**Discussion by maxima120** [sebastienros/jint#1526] [Q&A]
**Version used**

2.11.58

**Describe the bug**

Please state whether functionality of the package dependent on any specific software on the machine it is running. If yes then please enumerate all of them

**Additional context**

the jint.dll is 240Kb and I simply cant believe this is a standalone solution not dependent on any external Java VMs.

**Comment by lahma** [comment]
Jint is an interpreter which has no host machine dependencies, only libraries referenced by it are required. Java plays no role here, only Javascript.

You should probably concentrate on using 3.x beta as it's where the development push is.

**Comment by maxima120** [comment]
Thank you. Just clarify one thing for me please (ignoring my sloppy java vs js wording) - jint does not do any data marshaling to a native js runtime somewhere on my machine, evaluation / execution of the scripts stays inside **my** .net runtime and my OS process (I had quick look it seems jint converts js script into .net expression trees).. Is the above correct ?

**Reply by lahma** [reply]
No that I know of. There's no such dependencies.

**Comment by maxima120** [comment]
I just found mentioning of `Jint.Runtime.Interop;` in beta change log. Why would be there interop if everything is inside single .NET app?
@lahma

**Reply by lahma** [reply]
No idea.
